src/System.Private.ServiceModel/tools/setupfiles/BuildWCFSelfHostedService.cmd

```diff
@@ -0,0 +1,49 @@
+@echo off
+setlocal
+
+if not defined VisualStudioVersion (
+    if defined VS140COMNTOOLS (
+        call "%VS140COMNTOOLS%\VsDevCmd.bat"
+        goto :EnvSet
+    )
+
+    if defined VS120COMNTOOLS (
+        call "%VS120COMNTOOLS%\VsDevCmd.bat"
+        goto :EnvSet
+    )
+
+    echo Error: %~nx0 requires Visual Studio 2013 or 2015.
+    echo        Please see https://github.com/dotnet/wcf/wiki/Developer-Guide for build instructions.
+    exit /b 1
+)
+
+:EnvSet
+
+:: Log build command line
+set _buildproj1=%~dp0..\SelfHostedWcfService\SelfHostedWCFService.sln
+set _buildlog=%~dp0..\..\..\..\msbuildSelfHostedWcfService.log
+set _buildprefix=echo
+set _buildpostfix=^> "%_buildlog%"
+
+call :build %*
+
+:: Build
+set _buildprefix=
+set _buildpostfix=
+call :build %*
+
+goto :AfterBuild
```

Comment:
Unlike build scripts, we have no plan to sync this script with anything in the future. This is our script to maintain. It's okay to "borrow" code but no reason to let "bad" code spread. I know you have merged, but I feel strongly about this one (and similar situations in the future) :smiling_imp:.